src/engraving/rendering/dev/pagelayout.cpp

```diff
@@ -360,10 +361,22 @@ void PageLayout::collectPage(LayoutContext& ctx)
                                         Fingering* fingering = toFingering(e);
                                         if (fingering->isOnCrossBeamSide()) {
                                             TLayout::layoutFingering(fingering, fingering->mutldata());
+                                            if (fingering->addToSkyline()) {
+                                                const Note* n = fingering->note();
+                                                const RectF r
+                                                    = fingering->ldata()->bbox().translated(
+                                                          fingering->pos() + n->pos() + n->chord()->pos() + segment->pos()
+                                                          + segment->measure()->pos());
+                                                s->staff(fingering->note()->chord()->vStaffIdx())->skyline().add(r);
+                                            }
+                                            shapesToRecreate.insert(fingering->staffIdx());
                                         }
                                     }
                                 }
                             }
+                            for (const staff_idx_t staffIdx : shapesToRecreate) {
+                                segment->createShape(staffIdx);
```

Comment:
This looks good! Why is it necessary to recreate the segment shape though?

Reply:
Opps, correct - we don't!